### PR TITLE
[minor] Improve on the micro examples

### PR DESCRIPTION
**Pull request by blefaudeux**
## What does this PR do?
- microViT: fix the classification token option. Improves the classification a fair bit, and that's what ViT originally suggested
- microGPT: use the settings proposed in the [original minGPT char example](https://github.com/karpathy/minGPT/blob/master/play_char.ipynb). This slows down the training (roughly 40 minutes per epoch on a laptop) but is more of a apples-to-apples comparison
 
## Before submitting

- [ ] Did you have fun?
  - Make sure you had fun coding 🙃
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [ ] N/A
- [ ] Did you make sure to update the docs?
  - [ ] N/A
- [ ] Did you write any new necessary tests?
  - [ ] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/fairscale/blob/master/CHANGELOG.md)? (if needed)
  - [ ] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: examples/microViT.py [hunk 1]
```diff
@@ -151,11 +151,11 @@ def forward(self, x):
         x = x.flatten(2, 3).transpose(1, 2).contiguous()  # B HW C
 
         if self.hparams.classifier == Classifier.TOKEN:
-            # prepend classification token
+            # prepend classification token to the sequence
             clf_token = (
-                torch.ones(1, batch, self.hparams.dim, device=x.device) * self.clf_token
```
Comment (blefaudeux) [inline examples/microViT.py:156]: cc @SeanNaren, FYI the dimensions were broken here. We preprend the token to the sequence dimension, as suggested in the ViT paper

**Comment by blefaudeux** [comment]
Screenshot from 2021-11-29 22-18-40

File: examples/microViT.py (hunk 1 above)
Comment (SeanNaren) [reply examples/microViT.py:156]: thanks @blefaudeux for the headsup!

**Review by dianaml0 — APPROVED** [review]
LGTM!

**Pull request merged**